### PR TITLE
296 refactor list of authority signers should be optional

### PR DESCRIPTION
**Pull request by arminsabouri**
Resolves: https://github.com/botanix-labs/botanix/issues/296
and #297
Also, refactored optional fields to use a u8 bitmask. Alternatively, we could RLP encode/decode the whole thing but that may be space inefficient. And headers do need to be as lean as possible. 

lmk your thoughts on this.

### CODE REVIEW

**Comment by arminsabouri** [comment]
Drafting for now
Going to change poa to only add authorities during epoch blocks

**Comment by arminsabouri** [comment]
Good to go!

File: crates/botanix-lib/src/extra_data_header.rs [hunk 1]
```diff
@@ -205,12 +197,14 @@ impl ExtraDataHeader {
         };
 
         let signature = if optional_fields & (1u8 << HAS_SIGNATURE_POS) != 0 {
+            let recovery_id = RecoveryId::from_i32(i32::consensus_decode(reader)?).unwrap();
```
Comment (arminsabouri) [inline crates/botanix-lib/src/extra_data_header.rs:200]: remove unwrap!

**Review by scottmillner — APPROVED** [review]
🔥 🔥 🔥

**Pull request merged**